### PR TITLE
RD-479 list key values

### PR DESCRIPTION
**Pull request by YoniItzhak**
Implementing the `/labels/deployments/<key>` endpoint to list all values of a specific key in a tenant.

### CODE REVIEW

File: rest-service/manager_rest/rest/endpoint_mapper.py [hunk 1]
```diff
@@ -127,7 +127,8 @@ def setup_resources(api):
         'Sites': 'sites',
         'SitesName': 'sites/<string:name>',
         'ClusterStatus': 'cluster-status',
-        'DeploymentsLabels': 'labels/deployments'
+        'DeploymentsLabels': 'labels/deployments',
+        'DeploymnetsLabelsKey': 'labels/deployments/<string:key>'
```
Comment (tehasdf) [inline rest-service/manager_rest/rest/endpoint_mapper.py:131]:
```suggestion
        'DeploymentsLabelsKey': 'labels/deployments/<string:key>'
```

File: rest-service/manager_rest/rest/resources_v3_1/labels.py [hunk 2]
```diff
@@ -13,4 +13,17 @@ def get(self):
             models.DeploymentLabel.key).distinct().all()
 
         keys_list = [result.key for result in results]
-        return {'metadata': {}, 'items': keys_list}
+        return {'metadata': {'totatl': len(keys_list)}, 'items': keys_list}
+
+
+class DeploymnetsLabelsKey(SecuredResource):
```
Comment (tehasdf) [inline rest-service/manager_rest/rest/resources_v3_1/labels.py:19]:
```suggestion
class DeploymentsLabelsKey(SecuredResource):
```

File: rest-service/manager_rest/rest/resources_v3_1/__init__.py [hunk 3]
```diff
@@ -93,5 +93,6 @@
 )
 
 from .labels import (                           # NOQA
-    DeploymentsLabels
+    DeploymentsLabels,
+    DeploymnetsLabelsKey
```
Comment (tehasdf) [inline rest-service/manager_rest/rest/resources_v3_1/__init__.py:97]:
```suggestion
    DeploymentsLabelsKey
```

File: rest-service/manager_rest/rest/resources_v3_1/labels.py [hunk 4]
```diff
@@ -13,4 +13,17 @@ def get(self):
             models.DeploymentLabel.key).distinct().all()
 
         keys_list = [result.key for result in results]
-        return {'metadata': {}, 'items': keys_list}
+        return {'metadata': {'totatl': len(keys_list)}, 'items': keys_list}
```
Comment (tehasdf) [inline rest-service/manager_rest/rest/resources_v3_1/labels.py:16]:
```suggestion
        return {'metadata': {'total': len(keys_list)}, 'items': keys_list}
```

File: rest-service/manager_rest/rest/endpoint_mapper.py [hunk 5]
```diff
@@ -127,7 +127,8 @@ def setup_resources(api):
         'Sites': 'sites',
         'SitesName': 'sites/<string:name>',
         'ClusterStatus': 'cluster-status',
-        'DeploymentsLabels': 'labels/deployments'
+        'DeploymentsLabels': 'labels/deployments',
+        'DeploymentsLabelsKey': 'labels/deployments/<string:key>'
```
Comment (geokala) [inline rest-service/manager_rest/rest/endpoint_mapper.py:131]: Here, you dropped a ','

File: rest-service/manager_rest/rest/resources_v3_1/__init__.py [hunk 6]
```diff
@@ -93,5 +93,6 @@
 )
 
 from .labels import (                           # NOQA
-    DeploymentsLabels
+    DeploymentsLabels,
+    DeploymentsLabelsKey
```
Comment (geokala) [inline rest-service/manager_rest/rest/resources_v3_1/__init__.py:97]: More commas. Yes, I know it's a style choice- but if you'd done it on the previous PR you'd not need to show an extra +1-1 on the git changes on this commit.

File: rest-service/manager_rest/rest/resources_v3_1/labels.py [hunk 7]
```diff
@@ -10,12 +10,29 @@ class DeploymentsLabels(SecuredResource):
...
+        current_tenant = utils.current_tenant
         results = (db.session.query(models.DeploymentLabel.key)
                    .join(models.Deployment)
                    .filter(models.Deployment._tenant_id == current_tenant.id)
                    .distinct()
                    .all())
 
         keys_list = [result.key for result in results]
-        return {'metadata': {}, 'items': keys_list}
+        return {'metadata': {'total': len(keys_list)}, 'items': keys_list}
```
Comment (geokala) [inline rest-service/manager_rest/rest/resources_v3_1/labels.py:21]: Are we not using paging on this? I think the paging stuff has logic for injecting total, etc already?

File: rest-service/manager_rest/rest/endpoint_mapper.py (hunk 5 above)
Comment (YoniItzhak) [reply rest-service/manager_rest/rest/endpoint_mapper.py:131]: Thanks, will add it. I need to get used to it

File: rest-service/manager_rest/rest/resources_v3_1/__init__.py (hunk 6 above)
Comment (YoniItzhak) [reply rest-service/manager_rest/rest/resources_v3_1/__init__.py:97]: Right, changed too.

File: rest-service/manager_rest/rest/resources_v3_1/labels.py (hunk 7 above)
Comment (YoniItzhak) [reply rest-service/manager_rest/rest/resources_v3_1/labels.py:21]: I frankly don't know. @tehasdf ?

Comment (tehasdf) [reply rest-service/manager_rest/rest/resources_v3_1/labels.py:21]: of course we're not using pagination, that is only added by the storage-manager, in .list(), and this is a bare db.session.query

Comment (geokala) [reply rest-service/manager_rest/rest/resources_v3_1/labels.py:21]: Then do we need to add 'total', since it'll be added later?

**Review by tehasdf — CHANGES_REQUESTED** [review]
typo plz, same as #2458

**Review by tehasdf — DISMISSED** [review]

**Review by tehasdf — APPROVED** [review]

**Pull request merged**